src/ra_utility.h

```diff
@@ -172,6 +176,77 @@ _NODISCARD inline auto __cdecl tstrlen(_In_z_ LPCTSTR _Str) noexcept
 
 _END_EXTERN_C
 
+// Pro-tip: don't use C functions, if you must there must be a lot of checks done.
+// TODO: Check if an arg is nullptr, it's kind of complicated
+/// <summary>Writes output to <paramref name="buffer" /> with arguments specified.</summary>
+/// <typeparam name="CharT">
+///   A character type, specify this explicitly to use a different character set than configured.
+///   Valid character types are <c>char</c>, <c>signed char</c>, and <c>wchar_T</c>. Using any other
+///   type will disable this function.
+/// </typeparam>
+/// <param name="buffer">
+///   Pointer to a character string to write to, if this is <c>nullptr</c>, the operation is
+///   undefined.
+/// </param>
+/// <param name="bufsz">
+///   Up to <c><paramref name="bufsz" /> - 1</c> characters may be written, plus the null terminator.
+///   If the size is <c>0</c> or greater than <c>RSIZE_MAX</c>, the operation is undefined.
+/// </param>
+/// <param name="format">
+///   Pointer to a null-terminated multibyte string or wide-character string specifying how to
+///   interpret the data. The format string consists of ordinary multibyte characters(except %),
+///   which are copied unchanged into the output stream, and conversion specifications. The %n
+///   specifier is considered undefined.
+/// </param>
+/// <param name="...args">
+///   Forwarded arguments specifying data to print. If the number of arguments is more than the
+///   number of format specifiers present, the operation is undefined.
+/// </param>
+/// <returns>
+///   The number of characters written to <paramref name="buffer" /> excluding the null character.
+/// </returns>
+template<typename CharT = TCHAR, typename ...Args, typename = std::enable_if_t<is_char_v<CharT>>>
+_Success_(return >= 0) /*_NODISCARD*/ inline auto __cdecl
+stprintf_s(_Inout_updates_bytes_(bufsz)  CharT*       const __restrict buffer,
+           _In_range_(1, RSIZE_MAX)      rsize_t      const            bufsz,
+           _In_z_ _Printf_format_string_ const CharT* const __restrict format,
+           _In_                          Args&&...                     args)
```

Comment:
These pre/post conditions are only checked during code analysis, this is based off the the specification for C11's `sprintf_s`. MSVC version is not this strict.